micropip/wheelinfo.py

```diff
@@ -42,16 +40,15 @@ class WheelInfo:
     # Fields below are only available after downloading the wheel, i.e. after calling `download()`.
 
     _data: IO[bytes] | None = None  # Wheel file contents.
-    _dist: Distribution | None = None  # pkg_resources.Distribution object.
+    _metadata: Metadata | None = None  # Wheel metadata.
     _requires: list[Requirement] | None = None  # List of requirements.
 
-    # Note: `_project_name`` is taken from the wheel metadata, while `name` is taken from the wheel filename or metadata of the package index.
-    #       They are mostly the same, but can be different in some weird cases (e.g. a user manually renaming the wheel file), so just to be safe we store both.
-    _project_name: str | None = None  # Project name.
```

Comment:
So we would no longer get the name from the wheel metadata. then? I feel it's a safer source than the filename.

Reply:
> no longer get the name from the wheel metadata. then?

Interestingly, we didn't calculate the name from the metadata even before.

The `project_name` was calculated by just escaping the name that was passed by the caller.

https://github.com/pyodide/micropip/blob/1807ccb1a9241ee0023d59d5079a25c89378ad3f/micropip/externals/pip/_vendor/pkg_resources.py#L187

I moved this part to `__post_init__` method. So the behavior is not changed in this PR.

Reply:
I'll open a follow-up PR to parse the package name from the metadata file.